### ISSUE TITLE
Should create parent directories to create external files

### ISSUE THREAD

**Issue by BasixKOR** [twpayne/chezmoi#1570]
## Describe the bug
When you create new external file in `.chezmoiexternal` and try to apply it, chezmoi fails to apply with these message:

```
chezmoi: open C:/Users/basix/.vim/autoload/plug.vim: The system cannot find the path specified.
```

## To reproduce
```toml
[".vim/autoload/plug.vim"]
    type = "file"
    url = "https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim"
    refreshPeriod = "168h"
```

Make sure you don't have a .vim directory to reproduce this issue.

## Expected behavior

It should create directories as `mkdir -p` does.

## Output of command with the `--verbose` flag

```console
$ chezmoi --verbose apply
[literally the entire source code of vim-plug]
chezmoi: open C:/Users/basix/.vim/autoload/plug.vim: The system cannot find the path specified.
```

## Output of `chezmoi doctor`

```console
$ chezmoi doctor
RESULT   CHECK                MESSAGE
ok       version              v2.7.3, commit 32a15ac909ee8108d0eaeebcdbbac2305e32a9ba, built at 2021-10-23T00:31:10Z, built by goreleaser
ok       os-arch              windows/amd64
ok       executable           ~\scoop\apps\chezmoi\current\chezmoi.exe
ok       config-file          no config file found
ok       source-dir           C:/Users/basix/.local/share/chezmoi is a directory
ok       suspicious-entries   C:/Users/basix/.local/share/chezmoi: no suspicious entries
ok       working-tree         C:/Users/basix/.local/share/chezmoi is a directory
ok       dest-dir             C:/Users/basix is a directory
ok       shell                found C:\msys64\usr\bin\zsh.exe
ok       edit-command         found C:\Windows\System32\notepad.exe
ok       git-command          found C:\msys64\mingw64\bin\git.exe, version 2.33.0
ok       merge-command        found C:\msys64\usr\bin\vimdiff.exe
ok       age-command          age not found in $PATH
ok       gpg-command          found C:\msys64\usr\bin\gpg.exe, version 2.2.29
ok       1password-command    found ~\scoop\shims\op.exe, version 1.12.3
info     bitwarden-command    bw not found in $PATH
info     gopass-command       gopass not found in $PATH
info     keepassxc-command    keepassxc-cli not found in $PATH
info     keepassxc-db         not set
info     lastpass-command     lpass not found in $PATH
info     pass-command         pass not found in $PATH
ok       vault-command        found ~\scoop\shims\vault.exe, version 1.8.4
info     secret-command       not set
```

**Comment by twpayne** [comment]
Thanks for the clear report. This is the same issue as #1205.

**Comment by twpayne** [comment]
Thanks for the push on this #1571 should fix the issue.

**Comment by twpayne** [comment]
Re-opening this because this is actually a subset of #1205 which is fixed by #1571, but not all of #1205 is fixed.
